client/src/main/scala/com/redi/scalademo/presentation/FormListener.scala

```diff
@@ -0,0 +1,128 @@
+package com.redi.scalademo.presentation
+
+import autowire.{clientCallable, unwrapClientProxy}
+import com.redi.scalademo.business.{Calculator, NumericStringValidator, ValidatedResult, ValidationFailure}
+import com.redi.scalademo.infrastructure.Client
+import org.scalajs.dom
+import org.scalajs.jquery.{JQuery, JQueryEventObject, jQuery ⇒ $}
+
+import scala.concurrent.Future
+import scala.math.BigDecimal
+import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue
+import scala.scalajs.js
+
+class FormListener(
+  client: Client,
+  numericStringValidator: NumericStringValidator
+) {
+
+  private val InfoClass = "is-info"
+  private val DangerClass = "is-danger"
+
+  private val Validate = "validate"
+  private val Augend = "augend"
+  private val Addend = "addend"
+  private val Summand = "summand"
+
+  private val Name = "name"
+  private val Value = "value"
+
+  private val namedElement = "[name]"
+
+  private var formElement: JQuery = _
+  private var summandElement: JQuery = _
+
+  private var shouldValidate: Boolean = _
+  private var allFormValuesAreValid: Boolean = _
+  private var calculableFormData: js.Array[js.Dictionary[String]] = _
+  private var augend: String = _
+  private var addend: String = _
+
+  def attachTo(formId: String): Unit = {
+
+    formElement = $(s"form#$formId")
+    summandElement = formElement.find(elementNamed(Summand))
+
+    formElement.submit { (e: JQueryEventObject) ⇒
+      e.preventDefault()
+      init()
+      processValidationFormData()
+      validateAndExtractNumbers()
+      invokeAddition()
+    }
+
+    formElement.find(namedElement).on("input", { (self: dom.Element, e: JQueryEventObject) ⇒
+      $(self).removeClass(DangerClass)
+      summandElement
+        .value("")
+        .removeClass(InfoClass)
+    }: js.ThisFunction1[dom.Element, JQueryEventObject, js.Any])
+
+  }
+
+  private def elementNamed(name: String): String = {
+    s"""[name="$name"]"""
+  }
+
+  private def init(): Unit = {
+    shouldValidate = true
+    allFormValuesAreValid = true
+    calculableFormData = js.Array[js.Dictionary[String]]()
+    augend = ""
+    addend = ""
+  }
+
+  private def processValidationFormData(): Unit = {
+    formElement.find(namedElement).removeClass(DangerClass)
+    val serializedFormData = formElement.serializeArray().asInstanceOf[js.Array[js.Dictionary[String]]]
+    for (formControlData: js.Dictionary[String] ← serializedFormData) {
+      if (formControlData(Name) == Validate) {
+        shouldValidate = formControlData(Value).nonEmpty
```

Comment:
I considered that, but I think it's safe not to do that since we know this will only occur once.